packages/survey-core/src/validator.ts

```diff
@@ -1,13 +1,51 @@
 import { Base } from "./base";
-import { ISurveyErrorOwner, ISurvey } from "./base-interfaces";
+import { ISurveyErrorOwner, ISurvey, IElement, IQuestion } from "./base-interfaces";
 import { SurveyError } from "./survey-error";
 import { CustomError, RequreNumericError } from "./error";
 import { ILocalizableOwner, LocalizableString } from "./localizablestring";
 import { Serializer } from "./jsonobject";
 import { ConditionRunner } from "./conditions";
-import { Helpers } from "./helpers";
+import { HashTable, Helpers } from "./helpers";
 import { IValueGetterContext } from "./conditionProcessValue";
 
+export interface IValidationParams {
+  fireCallback: boolean;
+  isOnValueChanged?: boolean;
+  focusOnFirstError?: boolean;
+  firstErrorQuestion?: IQuestion;
+  callbackResult?: (res: boolean, element: IElement) => void;
+}
+export class AsyncElementsRunner {
```

Comment:
This class is called "runner". I expected to see a "run" function. May be it's better to call it something related to "await" functionality?